feat(search_toolkit):  Add Alibaba Tongxiao Search API Support

RonaldJEN opened the pull request:
# Add Alibaba Tongxiao Search API Support

## Feature Description

This PR introduces the `search_ali` method to the `SearchToolkit` class, integrating the Alibaba Tongxiao Search API. Tongxiao Search is a powerful real-time search API providing structured data from various search engines and knowledge bases, with specific optimizations for Chinese content search.

**API Reference:** [Standard Search API - GenericSearch](https://help.aliyun.com/document_detail/2857020.html?spm=a2c4g.11186623.0.0.4f79595fT05miJ)

## Implementation Details

- Retrieves API credentials via the `TONGXIAO_API_KEY` environment variable.
- Supports the following search parameters:
  - `timeRange`: Time frame filter (OneDay/OneWeek/OneMonth/OneYear/NoLimit)
  - `industry`: Industry filter (finance, law, medical, etc.)
  - `page`: Result pagination
  - `returnMainText`: Whether to return webpage main text
  - `returnMarkdownText`: Whether to return Markdown formatted content
  - `enableRerank`: Whether to enable result reranking (can reduce response time)

- Return Structure Formatting:
  - Extracts and standardizes key fields (title, snippet, url, etc.)
  - Prioritizes the `summary` field; uses `mainText` as a fallback summary if `summary` is unavailable.
  - Maintains consistency with the return structure of other search methods (e.g., `search_bing`).

## Documentation

The method includes a detailed docstring covering:
- Method description
- Parameter explanations
- Return value format
- Error handling mechanisms

This PR enhances CAMEL's search capabilities, particularly for Chinese content, offering users more diverse search options.

Wendong-Fan reviewed (commented):
thanks @RonaldJEN for the contribution! there's some conflicts, could you resolve this?

RonaldJEN commented:
@Wendong-Fan Thanks for pointing out the conflicts! I've resolved them now.

Wendong-Fan reviewed (commented):
thanks @RonaldJEN 's contribution, left some comments below, could we also add unit test code?

GitHoobar commented:
@RonaldJEN you can fix all things mentioned and its gtg

RonaldJEN commented:
@Wendong-Fan  I've completed all the requested changes:
1. Added query length validation (1-100 characters)
2. Changed `timeRange` to `time_range` and used `Literal` type
3. Applied `Literal` type to the `industry` parameter to restrict valid values
4. Converted all parameters from camelCase to snake_case
5. Standardized default value documentation using the format `(default: :obj:\`value\`)`
6. Simplified the return structure documentation
7. Changed response key names to snake_case for consistency

lightaime commented on camel/toolkits/search_toolkit.py:
```diff
@@ -1064,6 +1064,137 @@ def search_exa(
             return {"error": f"Exa search failed: {e!s}"}
 
+    @api_keys_required([(None, 'TONGXIAO_API_KEY')])
+    def search_ali(
```
Maybe make the name more specific:
```suggestion
    def search_alibaba_tongxiao(
```

GitHoobar commented:
@RonaldJEN can you resolve the conversation, which are fixed. Thanks

Wendong-Fan reviewed (commented):
we can also polish current docstring to highlight the feature difference between tongxiao and other search engine for better llm/agent understanding
